mayastor/src/bdev/nexus/nexus_share.rs

```diff
@@ -12,24 +13,109 @@ use crate::{
             CreateCryptoBdev,
             DestroyCryptoBdev,
             Error,
+            Error::AlreadyShared,
             Nexus,
             NexusTarget,
             ShareIscsiNexus,
             ShareNbdNexus,
             ShareNvmfNexus,
         },
-        nexus_iscsi::NexusIscsiTarget,
+        nexus_iscsi::{NexusIscsiError, NexusIscsiTarget},
         nexus_nbd::NbdDisk,
-        nexus_nvmf::NexusNvmfTarget,
+        nexus_nvmf::{NexusNvmfError, NexusNvmfTarget},
     },
-    core::Bdev,
+    core::{Bdev, Protocol, Share},
     ffihelper::{cb_arg, done_errno_cb, errno_result_from_i32, ErrnoResult},
 };
 
 /// we are using the multi buffer encryption implementation using CBC as the
 /// algorithm
 const CRYPTO_FLAVOUR: &str = "crypto_aesni_mb";
 
+#[async_trait(? Send)]
+///
+/// The sharing of the nexus is different compared to regular bdevs
+/// the Impl of ['Share'] handles this accordingly
+///
+/// The nexus and replicas are typically shared over different
+/// endpoints (not targets) however, we want to avoid to much
+/// iSCSI specifics and for bdevs the need for different endpoints
+/// is not implemented yet as the need for it has not arrived yet.
+impl Share for Nexus {
+    type Error = Error;
+    type Output = String;
+
+    async fn share_iscsi(&self) -> Result<Self::Output, Self::Error> {
+        match self.shared() {
+            Some(Protocol::Iscsi) => Ok(self.share_uri().unwrap()),
+            Some(p) => {
+                warn!("nexus {} already shared as {:?}", self.name, p);
```

Comment:
maybe this should be "error!"  because an error is returned? also for share_nvmf below.

Reply:
Yes, good catch, fixed!